.github/workflows/spread-tests.yaml

```diff
@@ -101,32 +131,41 @@ jobs:
     - name: Setup run tests variable
       if: "!contains(github.event.pull_request.labels.*.name, 'Skip spread')"
       run: |
+          get_new_tasks() {
+            local prefix=$1
+            local changes_param=$2
+            # The tests are just filtered when the change is a PR
+            # When 'Run Nested' label is added in a PR, all the nested tests have to be executed
+            TASKS_TO_RUN=""
+            if [ -z "${{ github.event.number }}" ] || [ "$RUN_NESTED" = 'true' ]; then
```

Comment:
I think we could also check here if -z "${{ inputs.rules }}"

Reply:
Mostly to make sure if we don't specify a rule set, this means we need to run all the tests defined in tasks

Reply:
Good point, done